domain-management/src/main/java/org/jboss/as/domain/management/security/SSLServerIdentityAddHandler.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2015, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.domain.management.security;
+
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.domain.management.logging.DomainManagementLogger;
+import org.jboss.dmr.ModelNode;
+
+/**
+ * Add handler for a child resource of a management security realm.
+ *
+ * @author Brian Stansberry (c) 2011 Red Hat Inc.
+ */
+public class SSLServerIdentityAddHandler extends SecurityRealmChildAddHandler {
+
+    public SSLServerIdentityAddHandler(boolean validateAuthentication, boolean validateAuthorization, AttributeDefinition... attributeDefinitions) {
+        super(validateAuthentication, validateAuthorization, attributeDefinitions);
+    }
+
+    @Override
+    protected void updateModel(OperationContext context, ModelNode operation) throws OperationFailedException {
+
+
+        ModelNode selfSignedCert = KeystoreAttributes.GENERATE_SELF_SIGNED_CERTIFICATE_HOST.resolveModelAttribute(context, operation);
+        if(selfSignedCert.isDefined()) {
+            ModelNode password = KeystoreAttributes.KEY_PASSWORD.resolveModelAttribute(context, operation);
+            if(!password.isDefined()) {
+                throw DomainManagementLogger.ROOT_LOGGER.noKeyPasswordForSelfSignedCert();
+            }
+        }
```

Comment:
Can't this just be "setRequires(ModelDescriptionConstants.KEY_PASSWORD)" on the GENERATE_SELF_SIGNED_CERTIFICATE_HOST definition builder?